clients/adapter/src/main/java/org/eclipse/hono/client/downstream/QoS.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+
+package org.eclipse.hono.client.downstream;
+
+
+/**
+ * A definition of the delivery semantics used for transferring messages.
+ *
+ */
+public enum QoS {
```

Comment:
Just stumbled upon this via https://github.com/eclipse/hono/pull/2216 : there's already a QoS class in `org.eclipse.hono.util` (and another one in `org.eclipse.hono.service.metric`). I think we could use the one in `org.eclipse.hono.util` here. It looks similar to me.

Reply:
nice catch :+1: I'll replace it with the `org.eclipse.hono.util.QoS` enum

Reply:
I thought you created a new one on purpose because they are actually quite different: `org.eclipse.hono.util.QoS` additionally contains `UNKNOWN`. I think you should set a valid QoS when invoking the client and I do not see value in opening the possibility of a new error case.

Reply:
@b-abel That might be true. However, I think we can also easily get rid of the existing `QoS.UNKNOWN` value as it doesn't add much value anyway.

Reply:
See #2233